docs/platform-engineers/addon/addon-cue.md

```diff
@@ -0,0 +1,407 @@
+metadata:
+  name: example
+  namespace: vela-system
+spec:
+  components: ...
+  policies:
+    - type: "topology"
+      name: "deploy-topology"
+      properties:
+        clusterLabelSelector: {}
+```
+
+Since an empty (`{}`) `clusterLabelSelector` topology will choose all exist clusters as target, so the components in application will be dispatched to all clusters including control-plane and managed-cluster.
+
+### Auxiliary resources
+
+You can also define some auxiliary resources in the outputs field of the `template.cue` file. These resources will only be applied to the control plane.
+
+```cue
+package main
+
+output: {
+	apiVersion: "core.oam.dev/v1beta1"
+	kind:       "Application"
+	metadata: {
+		name:      "example"
+		namespace: "vela-system"
+	}
+	spec: {
+		
+	}
+	... 
+}
+
+outputs: resourceTree: {
+	apiVersion: "v1"
+	kind:       "ConfigMap"
+	metadata: {
+		name:      "resource-tree"
+		namespace: "vela-system"
+		labels: {
+			"rules.oam.dev/resources":       "true"
+			"rules.oam.dev/resource-format": "json"
+		}
+	}
+	data: rules: json.Marshal(_rules)
+}
+
+_rules: {...}
+```
+
+In this example, we define a configmap `resourceTree` as an auxiliary resource, this configmap is actually a [resource topology rule](../../reference/topology-rule) . The function of this resource is to establish the relationship of CustomResources in the cluster, so that it can be displayed in the topology graph. It only needs to be applied to control-plane.
+
+You can also run the `cue eval *.cue resources/*.cue -e output -d yaml` command from local to see the result of resource rendering.
+
+### Use metadata of context to render application
+
+In addition to dynamically rendering the application by parameters, you can also read fields defined in `metadata.yaml` for rendering. For example, you can define a `template.cue` file as follows:
+
+```cue
+output: {
+	apiVersion: "core.oam.dev/v1beta1"
+	kind:       "Application"
+	metadata: {
+		name:      "example"
+		namespace: "vela-system"
+	}
+	spec: {
+		components: [
+			{
+				type: "webservice"
+				properties: {
+					image: "oamdev/vela-apiserver:" + context.metadata.version
+				}
+			},
+		]
+	}
+}
+
+```
+
+When rendering, the fields defined in `metadata.yaml` will be put into the CUE block of `context` and rendered together with other CUE files. For example, the `metadata.yaml` is:
+
+```yaml
+...
+name: velaux
+version: 1.2.4
+...
+```
+
+Resulting application is:
+
+```yaml
+apiVersion: core.oam.dev/v1beta1
+kind:       Application
+metadata:
+  name: example
+  namespace: "vela-system"
+spec:
+  components:
+    - type: webservice
+      properties:
+        image: "oamdev/vela-apiserver:v1.2.4"
+```
+
+The image tag becomes the addon's version due to the `context.metadata.version` points to. The real example is [VelaUX](https://github.com/kubevela/catalog/blob/master/addons/velaux/resources/apiserver.cue). Other available fields of metadata please refer to [metadata](./intro).
```

Comment:
The image tag becomes the addon's version which the `context.metadata.version` field points to.